tools/codegen/api/unboxing.py

```diff
@@ -0,0 +1,282 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+
+from tools.codegen.api import cpp
+from tools.codegen.api.cpp import argumenttype_type
+from tools.codegen.api.types import BaseCType, tensorOptionsT, CppSignature, CType, voidT
+from tools.codegen.model import (
+    Argument,
+    Type,
+    BaseType,
+    OptionalType,
+    ListType,
+    BaseTy,
+    TensorOptionsArguments,
+    NativeFunction,
+    Variant,
+)
+
+
+# This file generates the code for unboxing wrappers, i.e., the glue logic to unbox a boxed operator and convert the
+# ivalues from stack to correct arguments to the unboxed kernel, based on corresponding JIT schema. This codegen is
+# an alternative way to generate unboxing wrappers similar to the existing C++ metaprogramming approach but gets the
+# job done statically. These generated unboxing wrappers will be useful under the scenario where we need to register
+# a fixed set of operators known at compile time and thus can save some time in runtime initialization phase.
```

Comment:
I don't think "needing to register a fixed set of operators known at compile time" is the main reason for this - since you can already do that today with the templated unboxing, through selective build.

My understanding is that the codegen'd unboxing gives the mobile team more flexibility to extend it in various ways - decoupling it from the dispatcher and customizing it for mobile, e.g. by swapping out `IValue` for another (cheaper) boxed representation in the future.

Reply:
Yep I agree. I didn't understand dispatcher too well by the time I wrote it, will update it